src/init/features/dataconnect/sdk.ts

```diff
@@ -109,103 +106,90 @@ async function askQuestions(setup: Setup, config: Config): Promise<SDKInfo> {
-      );
-    const pkg =
-      newConnectorYaml.generate.javascriptSdk?.package ??
-      `@firebasegen/${connectorInfo.connectorYaml.connectorId}`;
-
     const javascriptSdk: JavascriptSDK = {
-      outputDir,
-      package: pkg,
-      packageJsonDir: newConnectorYaml.generate.javascriptSdk?.packageJsonDir,
+      outputDir: path.relative(
+        connectorDir,
+        path.join(appDir, `dataconnect-generated/js/${connectorYaml.connectorId}`),
+      ),
+      package: `@firebasegen/${connectorYaml.connectorId}`,
+      // If appDir has package.json, Emulator would install JS SDK for every reload.
+      // Otherwise, emulator would ignore it. Always add it here in case `package.json` is added later.
+      // TODO: Explore other platforms that can be automatically installed. Dart? Android?
+      packageJsonDir: path.relative(connectorDir, appDir),
     };
-
-    if (
-      (await directoryHasPackageJson(appDir)) &&
-      (await confirm({
-        message: "Would you like to add a dependency on the generated SDK to your package.json?",
-      }))
-    ) {
-      javascriptSdk.packageJsonDir = path.relative(connectorInfo.directory, appDir);
-    }
-    newConnectorYaml.generate.javascriptSdk = javascriptSdk;
+    connectorYaml.generate.javascriptSdk = javascriptSdk;
   }
 
   if (targetPlatform === Platform.DART) {
-    const outputDir =
-      newConnectorYaml.generate.dartSdk?.outputDir ||
-      path.relative(
-        connectorInfo.directory,
-        path.join(appDir, `generated/dart/${newConnectorYaml.connectorId}`),
-      );
-    const pkg = newConnectorYaml.generate.dartSdk?.package ?? newConnectorYaml.connectorId;
     const dartSdk: DartSDK = {
-      outputDir,
-      package: pkg,
+      outputDir: path.relative(
+        connectorDir,
+        path.join(appDir, `dataconnect-generated/dart/${connectorYaml.connectorId}`),
+      ),
+      package: connectorYaml.connectorId,
     };
-    newConnectorYaml.generate.dartSdk = dartSdk;
+    connectorYaml.generate.dartSdk = dartSdk;
   }
 
   if (targetPlatform === Platform.ANDROID) {
+    const kotlinSdk: KotlinSDK = {
+      outputDir: path.relative(connectorDir, path.join(appDir, `dataconnect-generated/kotlin`)),
+      package: `connectors.${snakeCase(connectorYaml.connectorId)}`,
+    };
     // app/src/main/kotlin and app/src/main/java are conventional for Android,
     // but not required or enforced. If one of them is present (preferring the
-    // "kotlin" directory), use it. Otherwise, fall back to the app directory.
-    let baseDir = path.join(appDir, `generated/kotlin`);
+    // "kotlin" directory), use it. Otherwise, fall back to the dataconnect-generated dir.
     for (const candidateSubdir of ["app/src/main/java", "app/src/main/kotlin"]) {
       const candidateDir = path.join(appDir, candidateSubdir);
-      if (fs.existsSync(candidateDir)) {
-        baseDir = candidateDir;
+      if (dirExistsSync(candidateDir)) {
```

Comment:
Had to change this to use the helper because `fs.existsSync` don't work well work mock-fs